kano/utils.py

```diff
@@ -69,6 +72,9 @@ def run_bg(cmd, localised=False):
     if not localised:
         env['LC_ALL'] = 'C'
 
+    if unsudo and 'SUDO_USER' in os.environ and os.environ['SUDO_USER'] != 'root':
+        cmd = "sudo -u {} bash -c '{}' ".format(os.environ['SUDO_USER'], cmd)
```

Comment:
@tombettany The logic is not quite the same as get_user_unsudoed(), because
we don't want to sudo for each run_cmd, only those for which we are
actually root. It may be that some logic can be shared though (if nothing
else, between the two locations you commented on).

On 20 May 2015 at 15:36, Tom <notifications@github.com> wrote:

> In kano/utils.py
> <https://github.com/KanoComputing/kano-toolset/pull/137#discussion_r30708296>
> :
>
> > @@ -69,6 +72,9 @@ def run_bg(cmd, localised=False):
> >      if not localised:
> >          env['LC_ALL'] = 'C'
> >
> > +    if unsudo and 'SUDO_USER' in os.environ and os.environ['SUDO_USER'] != 'root':
> > +        cmd = "sudo -u {} bash -c '{}' ".format(os.environ['SUDO_USER'], cmd)
>
> Likewise here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/KanoComputing/kano-toolset/pull/137/files#r30708296>.
>